apps/browser/src/content/autofill.js

```diff
@@ -632,13 +760,19 @@
           animateTheFilling = true;
 
       // Check if URL is not secure when the original saved one was
+      // START MODIFICATION
       function urlNotSecure(savedURLs) {
-          var passwordInputs = null;
           if (!savedURLs) {
               return false;
           }
 
-          return savedURLs.some(url => url.indexOf('https://') === 0) && 'http:' === document.location.protocol && (passwordInputs = document.querySelectorAll('input[type=password]'),
+          var passwordInputs = queryDocAll(document, document.body, function (el) {
+              return el.nodeName === 'INPUT' &&
+                  typeof el.attributes.type === 'string' &&
+                  el.attributes.type.toLowerCase() === 'password';
+          });
+
+          return savedURLs.some(url => url.indexOf('https://') === 0) && 'http:' === document.location.protocol,
               0 < passwordInputs.length && (confirmResult = confirm('Warning: This is an unsecured HTTP page, and any information you submit can potentially be seen and changed by others. This Login was originally saved on a secure (HTTPS) page.\n\nDo you still wish to fill this login?'),
                   0 == confirmResult)) ? true : false;
```

Comment:
This failed to compile for me. Here's how I fixed it, but please double check:

```suggestion
        return savedURLs.some(url =>
          url.indexOf('https://') === 0) &&
          'http:' === document.location.protocol &&
          0 < passwordInputs.length &&
          (confirmResult = confirm('Warning: This is an unsecured HTTP page, and any information you submit can potentially be seen and changed by others. This Login was originally saved on a secure (HTTPS) page.\n\nDo you still wish to fill this login?'),
            0 == confirmResult) ? true : false;
```

Reply:
Interesting, for me it did compile, though I didn't actually test this change and also wondered how it was written like this in the first place. I'll just take your suggested changes.

Reply:
Yes, and removed an extra `)` at the end. These lines are almost impossible to read.